sycl/include/sycl/properties/queue_properties.hpp

```diff
@@ -28,6 +28,21 @@ namespace property {
 namespace queue {
 class discard_events
     : public ::sycl::detail::DataLessProperty<::sycl::detail::DiscardEvents> {};
+
+class use_priority : public sycl::detail::PropertyWithData<
+                         sycl::detail::PropWithDataKind::QueuePriority> {
+public:
+  using priority_type = enum {
+    normal = 0, // default
+    low = 1,
+    high = 2
+  };
```

Comment:
This way users have to write like this (not duplicated "sycl::ext::oneapi::property::queue::use_priority"

``` C++
  sycl::property_list Props{
      sycl::ext::oneapi::property::queue::use_priority{
      sycl::ext::oneapi::property::queue::use_priority::low}};
```

I could instead change this to two `DataLessProperty` like 
```
sycl::ext::oneapi::property::queue::use_priority_low{}
sycl::ext::oneapi::property::queue::use_priority_high{}
```
What do you think?
Tag @gmlueck

Reply:
Yes, that seems less verbose.  Also, do we need the word "use"?  Can't we name these like:

```
sycl::ext::oneapi::property::queue::priority_low{}
sycl::ext::oneapi::property::queue::priority_high{}
```

How will you handle the case when both properties are specified for the same queue?  I think this should probably result in an exception when the queue is constructed.

Reply:
I prefer the named priorities, but we should keep the ability to specify the default.  This was possible with the parameter version, but got dropped; I think there should be a `sycl::ext::oneapi::property::queue::priority_normal` (or `priority_default`?) as well.

Reply:
Why do we need the `priority_default` since it will simply be ignored?

Reply:
I've just uploaded `priority_low` and `priority_high`

Reply:
> Why do we need the `priority_default` since it will simply be ignored?

Two reasons:
1) It's easier for people to write generic code and adjust the priorities of different kernels if they can always pass a priority to the queue constructor.
2) If a piece of code sets up many queues with priorities, it may improve readability to show the default priority explicitly.

Reply:
OK, added in https://github.com/intel/llvm/pull/7523/commits/ae196c6de3facf990945cdd643cecd6eaf1d198b